src/zeep/wsdl/messages/soap.py

```diff
@@ -116,6 +116,8 @@ def deserialize(self, envelope):
             return result
 
         result = result.body
+        if not hasattr(result, '__len__'): # Return body directly if len is allowed (could indicated valid primitive type).
```

Comment:
It seems the comment should be _Return body directly if len is **NOT** allowed_.